src/ingest.rs

```diff
@@ -901,25 +907,34 @@ async fn handle_data<T>(
                             StorageKey::builder()
                                 .start_key(&time_series.id)
                                 .end_key(timestamp)
+                                .build()
                         }
                         RawEventKind::OpLog => {
-                            let Ok(op_log) = bincode::deserialize::<OpLog>(&raw_event) else {
+                            let Ok(mut op_log) = bincode::deserialize::<OpLog>(&raw_event) else {
                                 err_msg = Some("Failed to deserialize OpLog".to_string());
                                 break;
                             };
-                            let agent_id = format!("{}@{source}", op_log.agent_name);
-                            StorageKey::builder()
-                                .start_key(&agent_id)
-                                .end_key(timestamp)
+                            op_log.sensor.clone_from(&source);
+                            raw_event.clone_from(&bincode::serialize(&op_log).unwrap());
```

Comment:
This unwrap can cause the termination of our module, and I think we need to try our best to avoid it.

Reply:
I'll change it.